toolkit/authentication/src/test/java/com/arcgismaps/toolkit/authentication/SafeSuspendableCancellableTest.kt

```diff
@@ -0,0 +1,134 @@
+     * When the block throws an exception
+     * Then the exception is propagated
+     * @since 200.8.1
+     */
+    @Test
+    fun throwsOnFailure() = runTest {
+        val ex = RuntimeException("boom")
+        var caught: Throwable? = null
+        try {
+            safeSuspendableCancellable<Unit> { _ ->
+                throw ex
+            }
+        } catch (t: Throwable) {
+            caught = t
+        }
+        assertThat(caught).isInstanceOf(ex::class.java)
+        assertThat(caught?.message).isEqualTo(ex.message)
+    }
+
+    /**
+     * Given a safeSuspendableCancellable block
+     * When the block attempts to resume after throwing an exception
+     * Then the resume is ignored and the exception is propagated
+     * @since 200.8.1
+     */
+    @Test
+    fun throwExceptionFirstThenReturnValueIsIgnored() = runTest {
+        var safeResult: String? = null
+        val res = runCatching {
+            safeResult = safeSuspendableCancellable<String> { cont ->
+                throw IllegalStateException("This should be called first")
+                cont.resumeSafely("late") // ignored
+            }
+        }
+        assertThat(res.exceptionOrNull()).isInstanceOf(IllegalStateException::class.java)
+        assertThat(res.exceptionOrNull()?.message).isEqualTo("This should be called first")
+        assertThat(safeResult).isNull()
+
+    }
+
+    /**
+     * Given a safeSuspendableCancellable block
+     * When the block attempts to resume multiple times
+     * Then only the first resume is effective
+     * @since 200.8.1
+     */
+    @Test
+    fun testIgnoresSubsequentSignals() = runTest {
+        val safeResult = safeSuspendableCancellable<String> { cont ->
+            cont.resumeSafely("first")
+            cont.resumeSafely("ignored") // ignored
+            cont.resumeSafely("also ignored") // ignored
+        }
+        assertThat(safeResult).isEqualTo("first")
+    }
+
+    /**
+     * Given a safeSuspendableCancellable block
+     * When the parent coroutine is cancelled
+     * Then the continuation resumes with cancellation
+     * And the cancellation handler is invoked
+     * @since 200.8.1
+     */
+    @Ignore("TODO() this test fails")
+    @Test
+    fun parentCancellationResultsInCancellationOutcome() = runTest {
+        val invoked = AtomicBoolean(false)
+        val deferred = async {
+            safeSuspendableCancellable<String> { cont ->
+                // simulate long operation; do not resume
+                cont.invokeOnCancellation {
+                    invoked.set(true)
+                }
+            }
+        }
+        deferred.cancel()
+        var caught: Throwable? = null
+        try {
+            deferred.await() // expect cancellation
+        } catch (t: Throwable) {
+            caught = t
+        }
+        assertThat(caught).isInstanceOf(CancellationException::class.java)
+        assertThat(invoked.get()).isTrue()
+    }
```

Comment:
Adding a little delay should do the trick:
```suggestion
    @Test
    fun parentCancellationResultsInCancellationOutcome() = runTest {
        val invoked = AtomicBoolean(false)
        val deferred = async {
            safeSuspendableCancellable<String> { cont ->
                // simulate long operation; do not resume
                cont.invokeOnCancellation {
                    invoked.set(true)
                }
            }
        }
        delay(100)
        deferred.cancel()
        var caught: Throwable? = null
        try {
            deferred.await() // expect cancellation
        } catch (t: Throwable) {
            caught = t
        }
        assertThat(caught).isInstanceOf(CancellationException::class.java)
        assertThat(invoked.get()).isTrue()
    }
```

Reply:
test passes 👍🏼